components/admin_console/schema_admin_settings.jsx

```diff
@@ -0,0 +1,418 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+import {FormattedHTMLMessage, FormattedMessage} from 'react-intl';
+
+import * as I18n from 'i18n/i18n.jsx';
+
+import {SettingsTypes} from 'utils/constants.jsx';
+import {formatText} from 'utils/text_formatting.jsx';
+import * as Utils from 'utils/utils.jsx';
+import RequestButton from 'components/admin_console/request_button/request_button';
+import LoadingScreen from 'components/loading_screen.jsx';
+import AdminSettings from 'components/admin_console/admin_settings.jsx';
+import BooleanSetting from 'components/admin_console/boolean_setting.jsx';
+import TextSetting from 'components/admin_console/text_setting.jsx';
+import DropdownSetting from 'components/admin_console/dropdown_setting.jsx';
+import MultiSelectSetting from 'components/admin_console/multiselect_settings.jsx';
+import RadioSetting from 'components/admin_console/radio_setting.jsx';
+import GeneratedSetting from 'components/admin_console/generated_setting.jsx';
+import UserAutocompleteSetting from 'components/admin_console/user_autocomplete_setting.jsx';
+import SettingsGroup from 'components/admin_console/settings_group.jsx';
+
+export default class SchemaAdminSettings extends AdminSettings {
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.schema !== this.props.schema) {
+            this.setState(this.getStateFromConfig(nextProps.config, nextProps.schema));
+        }
+    }
+
+    getConfigFromState(config) {
+        const schema = this.props.schema;
+
+        if (schema) {
+            if (!config[schema.id]) {
+                config[schema.id] = {};
+            }
+
+            const configSettings = config[schema.id];
+
+            const settings = schema.settings || [];
+            settings.forEach((setting) => {
+                configSettings[setting.key] = this.state[setting.key];
+            });
+        }
```

Comment:
I don't see the connection between `configSettings` and the return `config` value. Could you help me understand?